logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -218,16 +218,16 @@ def take_originals_from_queue(queue, size, wait)
           if event.nil?
             # queue poll timed out
             next
-          elsif event == LogStash::SHUTDOWN
+          elsif event.is_a?(LogStash::SignalEvent)
             # We MUST break here. If a batch consumes two SHUTDOWN events
             # then another worker may have its SHUTDOWN 'stolen', thus blocking
             # the pipeline.
-            @shutdown_signal_received = true
-            break
-          elsif event == LogStash::FLUSH
+            @shutdown_signal_received = event.shutdown?
```

Comment:
do not forget that that these ifs are called for all events. since signal are very **rare**, replacing the 2 ifs for both `LogStash::ShutdownEvent` and `LogStash::FlushEvent` for a single `if event.is_a?(LogStash::SignalEvent)` is more efficient and cleaner. Granted this is probably negligible and nonetheless, it is a useless if cascade.